### DISCUSSION TITLE
"Error: Unprocessable Entity" response for Tutorial -> FastAPI Body - Nested Models -> Bodies of arbitrary dicts

### DISCUSSION THREAD

**Discussion by azataiot** [fastapi/fastapi#6208] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI

app = FastAPI()


@app.post("/index-weights/")
async def create_index_weights(weights: dict[int, float]):
    return weights
```

### Description

- Just copy and run the sample code provided by official doc,
- It should return a `200` response for this request:
```json
{
  "test":{
    "1":"3.14"
  }
}
```
![CleanShot 2022-12-03 at 11 36 51](https://user-images.githubusercontent.com/8280770/205432208-f81af05d-6e48-496c-a282-b211ec1738bc.png)

### Operating System

macOS

### FastAPI Version

0.88.0

### Python Version

Python 3.10.8

### Additional Context

I have tried many other possible 'answers' to the endpoint but didn't worked.

**Comment by davismartens** [comment, marked as answer]
As per docs - "In this case, you would accept any `dict `as long as it has `int `keys with `float `values"

You have to pass in a dict of `int `and `float`. You are passing in a `dict `of `str`, `float` and `int`

For the example to work you need to pass:

Works:
`{
  "1" : 3.44
}`

Doesn't work
`{
  "one" : 3.44
}`


Please close the issue.
